src/page-modules/assistant/layout.tsx

```diff
@@ -106,7 +108,9 @@ function AssistantLayout({
         : undefined,
       transportModeFilter,
     );
-    router.push({ pathname: '/assistant', query });
+    router
+      .push({ pathname: '/assistant', query })
+      .then(() => setIsSearching(false));
```

Comment:
Could `await router.push()` instead, to keep it mostly consistent with the rest, using `async/await` syntax and not `Promise` syntax.